com.unity.netcode.gameobjects/Editor/Configuration/NetcodeForGameObjectsSettings.cs

```diff
@@ -41,41 +42,59 @@ internal static void SetAutoAddNetworkObjectSetting(bool autoAddSetting)
         }
     }
 
-    [FilePath("ProjectSettings/NetcodeForGameObjects.settings", FilePathAttribute.Location.ProjectFolder)]
-    internal class NetcodeForGameObjectsProjectSettings : ScriptableSingleton<NetcodeForGameObjectsProjectSettings>
+    [Serializable]
+    internal class NetcodeForGameObjectsJsonProjectSettings
```

Comment:
Is this temporary just for this release? I see you're merging to the release branch, I'm assuming we'll try to figure out what's going on later with develop and we won't backport this correct?

Reply:
I don't think that'd be a good plan. ScriptableSingleton stores in a different format. If we ship this, I think we have to keep it, otherwise we'll end up suddenly releasing a breaking change later that'll revert all their settings to the default.

If you think it's really worthwhile to try to figure out why ScriptableSingleton doesn't work, I can try to put some more time into it. But I'm of the mind that if it's that finnicky, I'd rather not use it at all. I already put a day into it and while it saves correctly, I couldn't find anything that would convince it to load what it saved... when I debugged the code that's supposed to load the file, it was loading some completely unrelated item, EditorFallbackWindow or something like that.

Reply:
I was asking cause I don't think NGO is the first package to have custom project settings. I see netcode for entities uses ScriptableSingleton as well for example. Do they do things differently than us to make this work? Can we get inspiration from other packages on how they handle this?